sources/packages/backend/apps/queue-consumers/src/processors/assessment/_tests_/cancel-application-assessment.processor.e2e-spec.ts

```diff
@@ -0,0 +1,186 @@
+import { ZBClient } from "zeebe-node";
+import { Job } from "bull";
+import { createMock } from "@golevelup/ts-jest";
+import { INestApplication } from "@nestjs/common";
+import { CancelAssessmentQueueInDTO } from "@sims/services/queue";
+import { QueueNames } from "@sims/utilities";
+import {
+  createTestingAppModule,
+  describeProcessorRootTest,
+} from "../../../../test/helpers";
+import { CancelApplicationAssessmentProcessor } from "../cancel-application-assessment.processor";
+import { DataSource, Repository } from "typeorm";
+import {
+  createFakeDisbursementOveraward,
+  saveFakeApplicationDisbursements,
+} from "@sims/test-utils";
+import {
+  ApplicationStatus,
+  DisbursementOveraward,
+  DisbursementSchedule,
+  DisbursementScheduleStatus,
+  StudentAssessment,
+} from "@sims/sims-db";
+
+describe(
+  describeProcessorRootTest(QueueNames.CancelApplicationAssessment),
+  () => {
+    let app: INestApplication;
+    let processor: CancelApplicationAssessmentProcessor;
+    let zbClientMock: ZBClient;
+    let appDataSource: DataSource;
+    let studentAssessmentRepo: Repository<StudentAssessment>;
+    let disbursementOverawardRepo: Repository<DisbursementOveraward>;
+    let disbursementScheduleRepo: Repository<DisbursementSchedule>;
+
+    beforeAll(async () => {
+      const { nestApplication, dataSource, zbClient } =
+        await createTestingAppModule();
+      app = nestApplication;
+      zbClientMock = zbClient;
+      appDataSource = dataSource;
+      studentAssessmentRepo = dataSource.getRepository(StudentAssessment);
+      disbursementOverawardRepo = dataSource.getRepository(
+        DisbursementOveraward,
+      );
+      disbursementScheduleRepo = dataSource.getRepository(DisbursementSchedule);
+      // Processor under test.
+      processor = app.get(CancelApplicationAssessmentProcessor);
+    });
+
+    beforeEach(() => {
+      jest.resetAllMocks();
+    });
+
+    it("Should cancel the assessment pending disbursements and rollback overawards when the cancelled application has overawards and also one sent and one pending disbursements.", async () => {
+      // Arrange
+      const workflowInstanceId = "2251799814028835";
+      // Application and disbursements.
+      const application = await saveFakeApplicationDisbursements(
+        appDataSource,
+        null,
+        {
+          applicationStatus: ApplicationStatus.Cancelled,
```

Comment:
maybe we can pass `complete` status here and overwrite the status to `cancelled` later in the test, so that, the below portions gets covered by the function `saveFakeApplicationDisbursements`, 
![image](https://user-images.githubusercontent.com/77353155/231517745-e0f5de6b-63b3-4952-b2dc-741a67fbe26f.png) thoughts?

Reply:
I am not sure if I am following. Please keep in mind that COE status is not being tested here, only `disbursementScheduleStatus`. Please let me know what I am missing 😉

Reply:
ya, I know COE status is not tested here, it's my opinion, not a blocker.

Reply:
@ann-aot maybe I am still missing the point here. I did not get exactly what your option was. Happy to talk if needed.

Reply:
Glad that we have the conversation. As per the chat, we will keep this one as it is.